src/vcpkg/configure-environment.cpp

```diff
@@ -0,0 +1,81 @@
+#include <vcpkg/base/basic_checks.h>
+#include <vcpkg/base/downloads.h>
+#include <vcpkg/base/system.print.h>
+#include <vcpkg/base/system.process.h>
+
+#include <vcpkg/archives.h>
+#include <vcpkg/commands.version.h>
+#include <vcpkg/configure-environment.h>
+#include <vcpkg/tools.h>
+#include <vcpkg/vcpkgcmdarguments.h>
+#include <vcpkg/vcpkgpaths.h>
+
+namespace vcpkg
+{
+    int run_configure_environment_command(const VcpkgPaths& paths, View<std::string> args)
+    {
+        print2(Color::warning, "vcpkg-ce ('configure environment') is experimental and may change at any time.\n");
+
+        auto& fs = paths.get_filesystem();
+        auto& download_manager = paths.get_download_manager();
+        auto node_path = paths.get_tool_exe(Tools::NODE);
+        Path node_root = node_path.parent_path();
+        auto node_modules = paths.root / "node_modules";
+        auto ce_path = node_modules / "vcpkg-ce";
+        if (!fs.is_directory(ce_path))
+        {
+            auto env = get_modified_clean_environment({}, node_root);
+#if defined(VCPKG_CE_SHA)
+            print2("Downloading vcpkg-ce bundle " VCPKG_BASE_VERSION_AS_STRING "\n");
+            const auto ce_uri =
+                "https://github.com/microsoft/vcpkg-tool/releases/download/" VCPKG_BASE_VERSION_AS_STRING
+                "/vcpkg-ce.tgz";
+            const auto ce_tarball = paths.downloads / "vcpkg-ce-" VCPKG_BASE_VERSION_AS_STRING ".tgz";
+            download_manager.download_file(fs, ce_uri, ce_tarball, std::string(MACRO_TO_STRING(VCPKG_CE_SHA)));
+#else  // ^^^ VCPKG_CE_BUNDLE_SHA / !VCPKG_CE_BUNDLE_SHA vvv
+            print2(Color::warning, "Downloading latest vcpkg-ce bundle\n");
+            const auto ce_uri = "https://github.com/microsoft/vcpkg-tool/releases/latest/download/vcpkg-ce.tgz";
+            const auto ce_tarball = paths.downloads / "vcpkg-ce-latest.tgz";
+            download_manager.download_file(fs, ce_uri, ce_tarball, nullopt);
+#endif // ^^^ !VCPKG_CE_BUNDLE_SHA
+            auto npm_path = node_root / "node_modules" / "npm" / "bin" / "npm-cli.js";
+            if (!fs.exists(npm_path, VCPKG_LINE_INFO))
+            {
+                npm_path = Path(node_root.parent_path()) / "lib" / "node_modules" / "npm" / "bin" / "npm-cli.js";
+            }
+
+            Command cmd_provision(node_path);
```

Comment:
Maybe we need a `.string_args(std::initializer_list<StringView>)`

Reply:
Since these are all constants `.raw_arg("--args --args --args")` would also make sense without such a code size penalty?

Reply:
SGTM as well; though shouldn't initializer_list be pretty cheap for code size? (note: not suggesting `template<class...Args>`)